### ISSUE TITLE
Better template model

### ISSUE THREAD

**Issue by chriswhawkins** [b2io/b2-styled-components#6]
Need to work out a better model for the way we handle templating out of the box. I'm working on a system something like you'd see in Sass, where I have a base component (e.g. Button) with default styles, some of which come from a function [think mixin] that interpolates the corresponding value from a theme or defers to a default if the theme provider isn't present. This allows for base components, with built in theming that can be easily set/changed and can also have additional context adjustments made via the theme provider.

**Comment by chriswhawkins** [comment]
@kalongthewires, did you have any opinions on how this is currently set up or the direction I'm going with it?

The "method" basically being what's currently in Button.js where I've got `Template` that interpolates the values from a ThemeProvider with some default fallbacks.

**Comment by kalongthewires** [comment]
I'm still thinking that `Template` should be a named mixin or a separate named component (e.g. Text) so it's explicitly reusable. We shouldn't generally need `Template` (i.e. mixin) functions inside a specific components file (e.g. Button.js). The mixin code should be placed a separate file in the styles folder (Mixins.js maybe?).

**Comment by chriswhawkins** [comment]
@kalongthewires I agree. Can we do that in a separate task (and possibly relate it to that theme/template open question issue I have)? Partly, because we'll have to go back and update that in all of the original examples (Button) and partly b/c this PR doesn't have a specific issue it's closing so it's not as easy to track.
